src/ServiceControl.Monitoring/Http/Diagrams/TimingsAggregator.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace ServiceControl.Monitoring.Timings
+{
+    class TimingsAggregator
+    {
+        public static List<MonitoredEndpoint> AggregateIntoLogicalEndpoints(ProcessingTimeStore processingTimeStore, CriticalTimeStore criticalTimeStore)
+        {
+            var now = DateTime.UtcNow;
+
+            var processingTimes = processingTimeStore.GetTimings(now);
+            var criticalTimes = criticalTimeStore.GetTimings(now);
+
+            var endpointNames = new[] { processingTimes, criticalTimes }
+                .SelectMany(t => t).Select(t => t.Id.EndpointName)
```

Comment:
Can the `SelectMany` and `Select` be collapsed into a single `SelectMany`? Either that or a `processingTimes.Concat(criticalTimes).Select(t => t.Id.EndpointName).Distinct()` could do the job.